frontend/ui/lib/widgets/create_trip_widget.dart

```diff
@@ -24,26 +24,26 @@ class _CreateTripWidgetState extends State<CreateTripWidget> {
   bool _enabled = false;
 
   String place;
-  int radius = 0;
-  String newInformation = 'Grabbed info placed here.';
-  void submitTrip() {
-    setState(() {
-      tripService.createTrip(Trip(
+  Future<Trip> submitTrip() async{
+    Trip trip = await tripService.createTrip(Trip(
```

Comment:
Nit: since you aren't doing anything else with the trip, you could just await and return the call to `createTrip` directly.

Even better, since the service already returns a `Future`, you can just return the result directly (and remove `async` from this function signature):

```dart
Future<Trip> submitTrip() {
  return tripService.createTrip(...);
}
```

Since it's only a single expression, I would say that you could even inline this code wherever `submitTrip` is used right now, but that may change down the line and it's nice to keep your callback wiring clean.